### PR TITLE
feat: paper-trade readiness — PT-FIX-25..27, 31..38, 40..44 (v0.32.0)

### PR DESCRIPTION
**Pull request by blahmonkey**
**NOTE:** The author must always **first review the PR themselves**, before requesting review from others, that means carefully having reviewed the full diff in GitHub.

## What does this PR change?

Closes 16 paper-trade readiness items (PT-FIX-25..27, 31..38, 40..44) surfaced across 10 iterative deep-diligence audit rounds against post-PT-FIX-1..24 state. Tracker: Vaquum/Praxis#77.

**Shutdown sequence (PT-FIX-25, 31, 36, 38, 44):** `ShutdownSequencer.shutdown()` now flips `state.mode` to HALTED before stopping any loop (`_halt_state_mode`); `_submit_exit` builds an `OrderContext` per shutdown EXIT and routes terminal outcomes through `OutcomeProcessor.process` so positions decrement; `_poll_until_terminal` accepts both PARTIAL and FILLED via `is_fill`, and routes pre-shutdown outcomes (queued before `_stop_outcome_loop` fired) through a new `non_pending_outcome_handler` callback the launcher wires to `process_outcome`. Defensive `.get()` lookup on `state.positions` prevents shutdown-aborting `KeyError` if a concurrent OutcomeLoop tick removed the position mid-shutdown.

**Validator bypass (PT-FIX-32, 33, 37):** `_should_bypass_stage` adds RISK and PRICE to the EXIT/ABORT/CANCEL bypass set (drawdown / stale orderbook / spread denials no longer block exits). `validate_intake_stage` exempts safety actions from `INTAKE_ORDER_NOTIONAL_ZERO`. New 16-case parametrized contract test pins the bypass set so future stage additions cannot silently miss the bypass list.

**Capital controller (PT-FIX-35, 40, 41, 43):** `reconcile_at_boot(positions=...)` resets stranded `reservation/in_flight/working` aggregates carried over from a crashed prior boot AND rebuilds `per_strategy_deployed` from live positions to prevent first-ENTER attribution-mismatch denial. `order_reject` and `order_cancel` now both accept IN_FLIGHT and WORKING states (handles venue REJECTED-after-ACK and EXPIRED-before-ACK races that previously parked capital permanently).

**Boot ordering (PT-FIX-26, 34):** `_determine_mode` defaults to REDUCE_ONLY when `health_snapshot` is unwired (closes the ~5s permissive window where `on_startup` ENTERs reached the venue before the first HealthLoop tick). `_wire_sensors` raises `StartupError` on a manifest with zero sensor specs instead of silently booting a permanently-dead account.

**HealthLoop race (PT-FIX-42):** `_apply_snapshot` re-checks `_running` under the lock before writing `state.mode`, so a tick already past the top-level `_running` guard cannot overwrite the PT-FIX-25 HALTED ratchet after `stop()` returned.

**Bridge helper + contract (PT-FIX-27):** `bridge_to_capital(controller, outcome)` centralises `CapitalController.send_order(reservation_id, command_id)` wiring; documents the `outcome.command_id ↔ send_order(order_id=...)` contract on `OutcomeProcessor.process`. Pre-fix every ACK/FILL returned `INVARIANT_BREACH: order not found` silently because no production caller invoked `send_order`.

**Greybeard pre-PR cleanups (last commit):** strict `ValueError` raise on dimensionally-wrong notional fallback; structured logs on each `bridge_to_capital` early-return; bypass-set contract test (#8 above).

**Tests:** 84 new (1182 total collected, 1174 passed + 8 skipped). Lint and mypy clean.

**Tech debt logged:** TD-027..033 in `docs/TechnicalDebt.md` for deferred edge cases (bare `assert` under `python -O`; loop `.stop()` cancel-but-no-join across all four loops; `_grow_position` defensive gap; `_poll_until_terminal` misleading double-dispatch; PraxisOutbound trade_id contract; `on_startup` capital_available gross-vs-net; multi-process WAL TOCTOU). All non-blocking for paper trade.

**Companion Praxis branch:** `feat/paper-trade-readiness-fixes` (Vaquum/Praxis) consumes this Nexus version via the `_NexusRuntime` wiring updates. The Praxis `pyproject.toml` `vaquum-nexus` pin will be bumped to this Nexus commit/tag once this PR merges.

## Checklist

- [x] I have reviewed full diff in "Files changed"
- [x] I left no unnecessary files in the changes
- [x] I ran `python -m pytest` locally without errors (1174 passing, 8 skipped)
- [ ] I updated `/docs` (if behavior/API/config/user/etc changed)
- [x] I added and/or updated docstrings as per [Writing Docstrings](https://github.com/Vaquum/dev-docs/blob/main/src/Writing-Docstrings.md) (for any changed public functions/classes)
- [x] I updated CHANGELOG.md
- [x] I updated pyproject.toml
- [x] I added and/or updated tests (if behavior changed or new code paths added)
- [x] I validated changes manually
- [x] I validated changes with LLM
- [x] I removed any extraneous examples/comments
- [ ] I linked issue to auto-close on merge (e.g., "Fixes #123") when applicable

### CODE REVIEW

File: nexus/core/capital_controller/capital_controller.py [hunk 1]
```diff
@@ -427,15 +543,21 @@ def order_reject(self, order_id: str) -> LifecycleResult:
                     category=FailureCategory.INVARIANT_BREACH,
                 )
 
-            if order.state != OrderLifecycleState.IN_FLIGHT:
+            if order.state == OrderLifecycleState.IN_FLIGHT:
+                self._state.in_flight_order_notional -= order.total
+            elif order.state == OrderLifecycleState.WORKING:
+                self._state.working_order_notional -= order.total
```
Comment (zero-bang) [inline nexus/core/capital_controller/capital_controller.py]: **Bug (defensive):** This should be `order.remaining_total`, not `order.total`.

`order_cancel` (line 705) correctly uses `remaining_total` for the WORKING path, accounting for partial fills that may have already decremented `working_order_notional` via `order_fill`. `order_reject` uses `total` for the same WORKING state.

For the PT-FIX-40 race (ACK-then-REJECT, no fills in between), `remaining_total == total` so this is a no-op today. But if a fill ever squeezes between ACK and a late REJECT, `total > remaining_total` and this line over-decrements `working_order_notional`, causing an accounting underflow. The same issue applies to `_adjust_strategy_deployed` on line 561.

Fix (mirrors `order_cancel` exactly):
```python
elif order.state == OrderLifecycleState.WORKING:
    self._state.working_order_notional -= order.remaining_total
```
And line 561:
```python
self._adjust_strategy_deployed(order.strategy_id, -order.remaining_total)
```

File: nexus/startup/shutdown_sequencer.py [hunk 2]
```diff
@@ -464,6 +580,106 @@ def _poll_until_terminal(
...
+        try:
+            self._non_pending_outcome_handler(outcome)
+        except Exception:  # noqa: BLE001 - handler failure must not abort shutdown
+            _log.exception(
+                'non_pending_outcome_handler raised during shutdown',
+                command_id=outcome.command_id,
+                outcome_type=outcome.outcome_type.value,
+            )
+
+    def _apply_terminal_outcome(self, outcome: object) -> None:
```
Comment (zero-bang) [inline nexus/startup/shutdown_sequencer.py]: Nit: `outcome` is typed as `object` but every call site passes a `TradeOutcome` (from `_poll_until_terminal` where it's the non-None return of `receive_outcome()`). Typing it as `TradeOutcome` would eliminate the three `# type: ignore` comments on lines 653-654 and 674, and preserve type-checker coverage for the `.command_id`, `.outcome_type`, and `.process(outcome, ...)` calls.

```python
def _apply_terminal_outcome(self, outcome: TradeOutcome) -> None:
```

**Review by zero-bang — CHANGES_REQUESTED** [review]
Thorough, well-documented PR with solid test coverage (84 new tests). The shutdown-halts-mode ratchet (PT-FIX-25), validator bypass expansions (PT-FIX-32/37), capital reconciliation (PT-FIX-35/41), and HealthLoop race fix (PT-FIX-42) are all correct and well-tested.

I traced control flow through `_poll_until_terminal` for every `TradeOutcomeType` variant (ACK, PARTIAL, FILLED, REJECTED, EXPIRED, CANCELED) and confirmed no double-dispatch or missed dispatch. The `bridge_to_capital` centralisation and the `OutcomeProcessor.process` contract doc are good additions.

One capital-accounting asymmetry needs fixing before merge — flagged inline. One minor typing suggestion also inline.

File: nexus/strategy/action_submit.py [hunk 3]
```diff
@@ -246,6 +254,72 @@ def submit_actions(
...
+    Args:
+        controller: Per-account `CapitalController` to mutate.
+        outcome: Result of a single action's `submit_actions` pass.
+
+    Returns:
+        `LifecycleResult` from `controller.send_order` when the
+        outcome carries a reservation and a `command_id`; `None`
+        when the outcome did not produce a tracked order (REJECTED,
+        SUBMIT_FAILED, INVALID, or ABORT — none of which reserved
+        capital that needs converting).
```
[the remark this answers is not in the document]
Comment (blahmonkey) [reply nexus/strategy/action_submit.py]: Addressed in 9714b13. Reworded the Returns docstring on `bridge_to_capital` to drop the inaccurate "or ABORT" listing and instead enumerate the three early-return cases explicitly: (1) status != SUBMITTED covers REJECTED/SUBMIT_FAILED/INVALID, (2) SUBMITTED but command_id is None as a defensive guard, (3) SUBMITTED with command_id but no reservation — that third case is where ABORT lands (validator bypassed → decision is None) alongside EXIT/MODIFY (validator runs but does not reserve capital).

File: nexus/startup/shutdown_sequencer.py [hunk 4]
```diff
@@ -464,6 +580,106 @@ def _poll_until_terminal(
...
+        '''Apply a shutdown-EXIT fill outcome to instance state.
+
+        PT-FIX-31: pre-fix the shutdown sequencer drained terminal
```
[the remark this answers is not in the document]
Comment (blahmonkey) [reply nexus/startup/shutdown_sequencer.py]: Addressed in 9714b13. Typed `_apply_terminal_outcome(self, outcome: TradeOutcome)` and removed both `# type: ignore[attr-defined]` (on `outcome.command_id` / `outcome.outcome_type`) and `# type: ignore[arg-type]` (on the `OutcomeProcessor.process` call). mypy now enforces the contract that callers must pass a `TradeOutcome`. `TradeOutcome` was already imported in this file so no new imports needed. mypy + ruff + 1174-test suite all clean post-fix.

File: nexus/startup/shutdown_sequencer.py [hunk 5]
```diff
@@ -161,6 +173,27 @@ def shutdown(self) -> None:
...
+        `state.mode != ACTIVE`, so flipping to HALTED here ensures
+        any in-flight outcome dispatch's downstream order is rejected
+        with `INTAKE_MODE_BLOCKS_ENTER` instead of leaking past
+        `_dispatch_shutdown` to the venue.
+        '''
+
+        self._state.mode = ModeState(
+            mode=OperationalMode.HALTED,
+            trigger='shutdown',
+            transitioned_at=datetime.now(timezone.utc),
```
[the remark this answers is not in the document]
Comment (blahmonkey) [reply nexus/startup/shutdown_sequencer.py]: Addressed in 9714b13. Changed to `datetime.now(tz=timezone.utc)` for consistency with the rest of the codebase (`nexus/core/health_loop.py:157`, etc.).

File: nexus/startup/sequencer.py [hunk 6]
```diff
@@ -660,16 +669,38 @@ def _determine_mode(self) -> None:
...
+            _log.warning(
+                'no health data available, defaulting to REDUCE_ONLY '
+                'until first HealthLoop tick',
+            )
+
+        if self._state is not None:
+            self._state.mode = ModeState(
+                mode=self._mode,
+                trigger=trigger,
+                transitioned_at=datetime.now(timezone.utc),
```
[the remark this answers is not in the document]
Comment (blahmonkey) [reply nexus/startup/sequencer.py]: Addressed in 9714b13. Changed `sequencer.py:702` to `datetime.now(tz=timezone.utc)` for consistency.

File: tests/test_startup_sequencer.py [hunk 7]
```diff
@@ -397,18 +397,67 @@ def test_wire_sensors_without_manifest_raises(self) -> None:
...
             sequencer._wire_sensors()
 
+    def test_wire_sensors_with_zero_sensor_specs_raises(self) -> None:
+        '''PT-FIX-34: pre-fix the all-sensors-failed guard short-
+        circuited on `attempted > 0`, so a manifest with strategies
+        but no sensor specs (`attempted == 0`) silently passed and
+        boot proceeded with `wired_sensors=[]`. The PredictLoop then
+        had no signal source and the account sat permanently dead.
+        Post-fix the guard fires whenever `_wired_sensors` is empty,
+        with a distinct error message.
```
[the remark this answers is not in the document]
Comment (blahmonkey) [reply tests/test_startup_sequencer.py]: Addressed in 9714b13. Reworded the docstring to accurately describe what `manifest.strategies = ()` exercises: no strategies → no sensor specs are ever attempted → `attempted == 0` → the empty `_wired_sensors` guard fires. The previous wording ("manifest with strategies but no sensor specs") was inaccurate because `Manifest`s validators reject both empty `strategies` and empty `sensors`, so that exact configuration cannot be constructed; the test models the equivalent state via the strategy-loop bypass.

File: nexus/core/capital_controller/capital_controller.py (hunk 1 above)
Comment (blahmonkey) [reply nexus/core/capital_controller/capital_controller.py]: Addressed in 9714b13. Switched all three sites to `order.remaining_total` matching the `order_cancel` and `order_fill` paths:

```python
if order.state == OrderLifecycleState.IN_FLIGHT:
    self._state.in_flight_order_notional -= order.remaining_total
elif order.state == OrderLifecycleState.WORKING:
    self._state.working_order_notional -= order.remaining_total
# ...
self._adjust_strategy_deployed(order.strategy_id, -order.remaining_total)
```

Pre-fix the math over-released by the cumulative filled portion when a partial-then-rejected order arrived (rare but possible on venue-side error after partial), causing the global `in_flight_order_notional`/`working_order_notional` and the per-strategy `deployed` to under-account for capital that was actually still parked at the venue. Existing tests stayed green; a future test should exercise PARTIAL → REJECT specifically.

File: nexus/startup/shutdown_sequencer.py (hunk 2 above)
Comment (blahmonkey) [reply nexus/startup/shutdown_sequencer.py]: Addressed in 9714b13 (same fix as the Copilot duplicate above). `_apply_terminal_outcome` is now typed as `(self, outcome: TradeOutcome) -> None` and both `# type: ignore[attr-defined]` (on `outcome.command_id` / `outcome.outcome_type`) and `# type: ignore[arg-type]` (on the `OutcomeProcessor.process` call) are gone. mypy now enforces the contract.

File: nexus/startup/shutdown_sequencer.py [hunk 8]
```diff
@@ -321,13 +354,87 @@ def _submit_exit(self, strategy_id: str, action: Action) -> None:
...
+                f'action.size={action.size}; refusing to construct OrderContext'
+            )
+            raise ValueError(msg)
+
+        return OrderContext(
+            command_id=command_id,
+            strategy_id=strategy_id,
+            trade_id=action.trade_id,
+            side=validation_context.order_side,
+            order_size=action.size,
```
[the remark this answers is not in the document]
Comment (blahmonkey) [reply nexus/startup/shutdown_sequencer.py:438]: Addressed in 82f3452. The diagnosis is correct — `OrderContext.is_entry` was deriving intent from venue side (`side == BUY`), so a short-position EXIT (BUY-to-cover) would have been routed through `OutcomeProcessor`s entry path: `_grow_position` instead of `_reduce_position`, plus a `capital.order_fill` lookup against an order id that was never registered via `bridge_to_capital`.

Fix takes the explicit-encoding option you preferred:

- Added required `is_entry: bool` field to `OrderContext`. `side` documents venue direction only; intent is carried by `is_entry`. `is_exit` becomes `not self.is_entry`. `__post_init__` rejects non-bool.
- `shutdown_sequencer._build_exit_order_context` now passes `is_entry=False`.
- All Nexus `OrderContext` fixtures updated; new tests cover short ENTER (`side=SELL, is_entry=True`) and short EXIT (`side=BUY, is_entry=False`) to lock in that the field replaces the BUY/SELL heuristic.
- Praxis launcher fix lands in a separate commit on the Praxis branch (`Vaquum/Praxis@4415899`): `_build_order_context` now passes `is_entry=action.action_type == ActionType.ENTER`. Both repos green: Nexus 1177/8 skipped, Praxis 925/1 skipped.

Note on present-day exposure: the runtime intake validator currently rejects `ENTER SELL` / `EXIT BUY` with `INTAKE_SPOT_DIRECTION_INVALID`, so shorts can only enter the system via the boot reconciler taking `Position.side` from venue truth. That made this latent today, but it would have surfaced once the bypass set widened or a short-side strategy was permitted — better to fix the OrderContext shape now.

File: tests/test_health_loop.py [hunk 9]
```diff
@@ -176,3 +176,81 @@ def test_start_is_idempotent() -> None:
...
+    loop.stop()
+
+    sentinel_trigger = 'sentinel-shutdown-ratchet'
+    state.mode.trigger = sentinel_trigger
+
+    release_snapshot.set()
+
+    import time
+    time.sleep(0.2)
+
```
[the remark this answers is not in the document]
Comment (blahmonkey) [reply tests/test_health_loop.py]: Addressed in 0fa8eda. Replaced the `time.sleep(0.2)` with a deterministic `threading.Timer.join(timeout=5.0)` on the in-flight tick:

```python
in_flight_timer = loop._timer
assert in_flight_timer is not None
loop.stop()
# ... set sentinel, release snapshot ...
in_flight_timer.join(timeout=5.0)
if in_flight_timer.is_alive():
    pytest.fail("in-flight tick did not complete within 5s after release_snapshot.set()")
```

`stop()` nulls `loop._timer`, so the reference is captured BEFORE `stop()`. The `Timer` thread runs `_tick()` (which calls `_apply_snapshot`) — `cancel()` does not preempt a running tick, so `join()` returns only after `_tick` and therefore `_apply_snapshot` fully complete. The trigger assertion now runs on a deterministically-quiescent loop instead of a sleep-grace window, so a regression of PT-FIX-42 fails reliably under CI load (and a stuck tick hard-fails on the join timeout rather than masquerading as a flaky sleep race).

Considered the wrapped-evaluator-with-`tick_done`-event option but it fires before the `_apply_snapshot` lock-acquire-and-check, so it would still need a bounded poll to be safe — joining the actual worker thread is the cleanest deterministic signal. The cost is reading one private attribute (`loop._timer`), which is acceptable since the test is specifically exercising HealthLoop`s internal in-flight-tick race.

File: nexus/startup/shutdown_sequencer.py [hunk 10]
```diff
@@ -454,8 +562,17 @@ def _poll_until_terminal(
...
+                self._dispatch_non_pending_outcome(outcome)
+                continue
+
+            if outcome.outcome_type.is_fill:
+                self._apply_terminal_outcome(outcome)
+
+            if outcome.outcome_type.is_terminal:
                 remaining.discard(outcome.command_id)
+                if not outcome.outcome_type.is_fill:
+                    self._apply_terminal_outcome(outcome)
```
[the remark this answers is not in the document]
Comment (blahmonkey) [reply nexus/startup/shutdown_sequencer.py]: Addressed in 104a63c. Removed the inner `if not outcome.outcome_type.is_fill: self._apply_terminal_outcome(outcome)` call — it was a no-op because `_apply_terminal_outcome` early-returns on `not outcome_type.is_fill`. Kept the call on the outer fill branch, which actually does the work (FILLED/PARTIAL → `OutcomeProcessor.process` → `_reduce_position`).

The contract — non-fill terminals (REJECTED / CANCELED / EXPIRED with no preceding PARTIAL) are deliberately NOT applied during shutdown because the venue did not close the position, so `state.positions` must remain untouched — is already documented on `_apply_terminal_outcome`. The duplicate call was just misleading dead code that would rot on the next edit, exactly as you flagged. Suite still 1177/8 skipped.

File: nexus/core/capital_controller/capital_controller.py [hunk 11]
```diff
@@ -53,6 +55,113 @@ def __init__(self, capital_state: CapitalState) -> None:
...
+                    'reconcile_at_boot resetting stranded working_order_notional: %s',
+                    stranded_working,
+                )
+                self._state.working_order_notional = _ZERO
+
+            if positions is not None:
+                rebuilt: dict[str, Decimal] = {}
+                for pos in positions:
+                    if pos.strategy_id is None:
+                        continue
```
[the remark this answers is not in the document]
Comment (blahmonkey) [reply nexus/core/capital_controller/capital_controller.py]: Addressed in 68ee0ff. Removed the `if pos.strategy_id is None` branch — confirmed unreachable since `Position.strategy_id: str` and `Position.__post_init__` rejects empty/whitespace via the `("trade_id", "strategy_id", "symbol")` loop. Letting the existing invariants speak for themselves; if a strategy-attribution gap ever needs to be tolerated, the right move is an explicit Optional shape with a logged skip — not a silent guard that masks data-quality issues.

File: nexus/startup/shutdown_sequencer.py [hunk 12]
```diff
@@ -321,13 +354,88 @@ def _submit_exit(self, strategy_id: str, action: Action) -> None:
...
+                validation_context=context,
+            )
+        except ValueError:
+            _log.exception(
+                'shutdown exit OrderContext rejected by invariants',
+                strategy_id=strategy_id,
+                trade_id=action.trade_id,
+                command_id=returned_id,
+            )
+
```
[the remark this answers is not in the document]
Comment (blahmonkey) [reply nexus/startup/shutdown_sequencer.py]: Addressed in 54cbd7b. Reordered `_submit_exit` to call `_build_exit_order_context` BEFORE appending to `_submitted_command_ids`. On ValueError we log loudly (with explicit copy explaining the tradeoff) and return early — no wait, no escalation, no noise.

Tradeoff acknowledged in the log: the venue may leave the order open since we skip the escalation-abort path. The recovery is two-stage:
1. Next boot`s `reconcile_at_boot` (PT-FIX-35/41) resets stranded `in_flight_order_notional` / `working_order_notional` / `reservation_notional` and rebuilds `per_strategy_deployed` from live positions.
2. A venue reconciliation pass (currently a TD entry — TD-027 in `Vaquum/Praxis docs/TechnicalDebt.md`) should detect and clean up the dangling order.

Updated `TestSubmitExitMissingPositionRace::test_position_removed_between_context_and_order_context_does_not_abort` to assert the new behavior (`praxis_cmd_42 not in _submitted_command_ids`) and added a paragraph to the test class docstring describing the change. Suite still 1177/8 skipped.

**Review by zero-bang — APPROVED** [review]
